### PR TITLE
Bug 1262219 - Get dialer app working as chrome r=gsvelto

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4a83fb5e5ecd6e6308efdd39d5f5f7dae3be2a7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=00d31ca76910e0c029155094981c768f81878c07

File: apps/dialer/js/handled_call.js [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+  navigator.mozL10n.setAttributes(this.durationChildNode, 'callEnded');
+
+  // FIXME/bug 1007148: Refactor duration element structure. No number or ':'
+  // existence checking will be necessary.
+  var totalDuration = !!currentDuration.match(/\d+/g) ? currentDuration : '';
+  this.totalDurationNode.textContent = totalDuration;
+
+  setTimeout(function(evt) {
+    CallScreen.removeCall(self.node);
+    CallsHandler.end();
```
Comment (gabrielesvelto) [inline apps/dialer/js/handled_call.js]: Why did you add the `end()` call here? In theory `hc_remove()` should be called only when a call has already ended so I think this is redundant.

File: apps/dialer/js/dialer.js [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+    } else if (data.type === 'show-navbar') {
+      NavbarManager.show();
+    }
+  }
+
+  window.addEventListener('message', handleMessage);
+
+  /* === Calls === */
+  function call(number, cardIndex) {
+    if (engineeringModeKey && number === engineeringModeKey) {
```
Comment (gabrielesvelto) [inline apps/dialer/js/dialer.js]: You can kill the engineering mode key support and all the associated tests if you feel like it: I've personally removed that API from gecko :-)

File: apps/dialer/js/handled_call.js (hunk 1 above)
Comment (albertopq) [inline apps/dialer/js/handled_call.js]: Oops, good catch. Just debugging

File: apps/dialer/js/dialer.js (hunk 2 above)
Comment (albertopq) [inline apps/dialer/js/dialer.js]: WIll do! :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1d89a0f76630566798e3faab84828795b4bd5c76

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bb21d676cde603fd70c7104583bbe03c813ad743
